public/components/trace_analytics/components/common/plots/service_map_node_details.tsx

```diff
@@ -18,7 +18,7 @@ import { ServiceNodeDetails } from '../../../../../../common/types/trace_analyti
 interface ServiceMapNodeDetailsProps {
   selectedNodeDetails: ServiceNodeDetails | null;
   setSelectedNodeDetails: React.Dispatch<React.SetStateAction<ServiceNodeDetails | null>>;
-  addServiceFilter: (selectedServiceName: string) => void;
+  addServiceFilter: (selectedServiceName: string | null) => void;
```

Comment:
are we expecting to have a null passed in for addServiceFilter?

Reply:
No, added in the null check to handle switching between custom and data prepper to clear selections and prevent existing bugs where the page would fail to render.